lib/shelf_io.dart

```diff
@@ -31,6 +31,52 @@ import 'src/util.dart';
 
 export 'src/io_server.dart' show IOServer;
 
+/// Starts an [HttpServer] that listens on the specified [address] and
+/// [port] and sends requests to [handler].
+///
+/// If a [securityContext] is provided an HTTPS server will be started.
+///
+/// Serves requests in an error zone, using [onError] to handle errors, if
+/// provided. Uses a default error handler that prints error summary if
+/// [onError] is not provided.
+///
+/// See the documentation for [HttpServer.bind] and [HttpServer.bindSecure]
+/// for more details on [address], [port], [backlog], and [shared].
+Future<HttpServer> serveGuarded(
+  Handler handler,
+  address,
+  int port, {
+  SecurityContext? securityContext,
+  int? backlog,
+  bool shared = false,
+  void Function(Object, StackTrace) onError = _logServeRequestErrors,
+}) async {
+  backlog ??= 0;
+  var server = await (securityContext == null
+      ? HttpServer.bind(address, port, backlog: backlog, shared: shared)
+      : HttpServer.bindSecure(
+          address,
+          port,
+          securityContext,
+          backlog: backlog,
+          shared: shared,
+        ));
+  serveRequestsGuarded(server, handler, onError: onError);
+  return server;
+}
+
+/// Serve a [Stream] of [HttpRequest]s with error handling.
+///
+/// Serves requests in an error zone, using [onError] to handle errors, if
+/// provided. Uses a default error handler that prints error summary if
+/// [onError] is not provided.
+void serveRequestsGuarded(Stream<HttpRequest> requests, Handler handler,
+    {void Function(Object, StackTrace) onError = _logServeRequestErrors}) {
+  return Chain.capture(() {
```

Comment:
I don't think we want to use `Chain.capture` for this since it comes with overhead.

Reply:
Interesting, I can add a flag for skipping the capture. Unfortunately traces are not very useful without it - how much is the overhead?